README.rst
Python Client for GKE Connect Gateway API API
=============================================

|preview| |pypi| |versions|

`GKE Connect Gateway API API`_: 

- `Client Library Documentation`_
- `Product Documentation`_

.. |preview| image:: https://img.shields.io/badge/support-preview-orange.svg
   :target: https://github.com/googleapis/google-cloud-python/blob/main/README.rst#stability-levels
.. |pypi| image:: https://img.shields.io/pypi/v/google-cloud-gke-connect-gateway.svg
   :target: https://pypi.org/project/google-cloud-gke-connect-gateway/
.. |versions| image:: https://img.shields.io/pypi/pyversions/google-cloud-gke-connect-gateway.svg
   :target: https://pypi.org/project/google-cloud-gke-connect-gateway/
.. _GKE Connect Gateway API API: https://cloud.google.com/anthos/multicluster-management/gateway
.. _Client Library Documentation: https://cloud.google.com/python/docs/reference/connectgateway/latest
.. _Product Documentation:  https://cloud.google.com/anthos/multicluster-management/gateway

Quick Start
-----------

In order to use this library, you first need to go through the following steps:

1. `Select or create a Cloud Platform project.`_
2. `Enable billing for your project.`_
3. `Enable the GKE Connect Gateway API API.`_
4. `Setup Authentication.`_

.. _Select or create a Cloud Platform project.: https://console.cloud.google.com/project
.. _Enable billing for your project.: https://cloud.google.com/billing/docs/how-to/modify-project#enable_billing_for_a_project
.. _Enable the GKE Connect Gateway API API.:  https://cloud.google.com/anthos/multicluster-management/gateway
.. _Setup Authentication.: https://googleapis.dev/python/google-api-core/latest/auth.html

Installation
~~~~~~~~~~~~

Install this library in a `virtualenv`_ using pip. `virtualenv`_ is a tool to
create isolated Python environments. The basic problem it addresses is one of
dependencies and versions, and indirectly permissions.

With `virtualenv`_, it's possible to install this library without needing system
install permissions, and without clashing with the installed system
dependencies.

.. _`virtualenv`: https://virtualenv.pypa.io/en/latest/


Code samples and snippets
~~~~~~~~~~~~~~~~~~~~~~~~~

Code samples and snippets live in the `samples/` folder.


Supported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^
Our client libraries are compatible with all current [active](https://devguide.python.org/devcycle/#in-development-main-branch) and [maintenance](https://devguide.python.org/devcycle/#maintenance-branches) versions of
Python.

Python >= 3.7

Unsupported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^^^
Python <= 3.6

If you are using an [end-of-life](https://devguide.python.org/devcycle/#end-of-life-branches)
version of Python, we recommend that you update as soon as possible to an actively supported version.


Mac/Linux
^^^^^^^^^

.. code-block:: console

    pip install virtualenv
    virtualenv <your-env>
    source <your-env>/bin/activate
    <your-env>/bin/pip install google-cloud-gke-connect-gateway


Windows
^^^^^^^

.. code-block:: console

    pip install virtualenv
    virtualenv <your-env>
    <your-env>\Scripts\activate
    <your-env>\Scripts\pip.exe install google-cloud-gke-connect-gateway

Next Steps
~~~~~~~~~~

-  Read the `Client Library Documentation`_ for GKE Connect Gateway API API
   to see other available methods on the client.
-  Read the `GKE Connect Gateway API API Product documentation`_ to learn
   more about the product and see How-to Guides.
-  View this `README`_ to see the full list of Cloud
   APIs that we cover.

.. _GKE Connect Gateway API API Product documentation:  https://cloud.google.com/anthos/multicluster-management/gateway
.. _README: https://github.com/googleapis/google-cloud-python/blob/main/README.rst


CHANGELOG.md
# Changelog

### [0.2.2](https://www.github.com/googleapis/python-gke-connect-gateway/compare/v0.2.1...v0.2.2) (2021-07-29)


### Features

* add Samples section to CONTRIBUTING.rst ([#15](https://www.github.com/googleapis/python-gke-connect-gateway/issues/15)) ([d3d7206](https://www.github.com/googleapis/python-gke-connect-gateway/commit/d3d72060bb9f2b1ea516b904b9118a66bf0ca209))


### Bug Fixes

* enable self signed jwt for grpc ([#19](https://www.github.com/googleapis/python-gke-connect-gateway/issues/19)) ([9091e3f](https://www.github.com/googleapis/python-gke-connect-gateway/commit/9091e3f97a5a0e6bd844243dba05093c6c09f188))


### Miscellaneous Chores

* release as 0.2.2 ([#20](https://www.github.com/googleapis/python-gke-connect-gateway/issues/20)) ([6494fae](https://www.github.com/googleapis/python-gke-connect-gateway/commit/6494fae2a5fe3e1d1472048fb889c78a520764eb))

### [0.2.1](https://www.github.com/googleapis/python-gke-connect-gateway/compare/v0.2.0...v0.2.1) (2021-07-21)


### Bug Fixes

* **deps:** pin 'google-{api,cloud}-core', 'google-auth' to allow 2.x versions ([#14](https://www.github.com/googleapis/python-gke-connect-gateway/issues/14)) ([3aa8b31](https://www.github.com/googleapis/python-gke-connect-gateway/commit/3aa8b3104cb2aa784556cd2896e9cf791177ef25))

## [0.2.0](https://www.github.com/googleapis/python-gke-connect-gateway/compare/v0.1.0...v0.2.0) (2021-07-14)


### Features

* add always_use_jwt_access ([#6](https://www.github.com/googleapis/python-gke-connect-gateway/issues/6)) ([0f73678](https://www.github.com/googleapis/python-gke-connect-gateway/commit/0f73678e2d73852985149d4ecdb0ee539d465374))


### Bug Fixes

* disable always_use_jwt_access ([#10](https://www.github.com/googleapis/python-gke-connect-gateway/issues/10)) ([2408f62](https://www.github.com/googleapis/python-gke-connect-gateway/commit/2408f6275b2dbd2d6db462c764577a9e2f2d5e59))

## 0.1.0 (2021-06-20)


### Features

* generate v1beta1 ([3a3b952](https://www.github.com/googleapis/python-gke-connect-gateway/commit/3a3b95276defd2d37b157179b4fb4ab728c3af9f))


owl-bot-staging/v1beta1/noxfile.py
# -*- coding: utf-8 -*-
# Copyright 2020 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#
import os
import pathlib
import shutil
import subprocess
import sys


import nox  # type: ignore

CURRENT_DIRECTORY = pathlib.Path(__file__).parent.absolute()

LOWER_BOUND_CONSTRAINTS_FILE = CURRENT_DIRECTORY / "constraints.txt"
PACKAGE_NAME = subprocess.check_output([sys.executable, "setup.py", "--name"], encoding="utf-8")


nox.sessions = [
    "unit",
    "cover",
    "mypy",
    "check_lower_bounds"
    # exclude update_lower_bounds from default
    "docs",
]

@nox.session(python=['3.6', '3.7', '3.8', '3.9'])
def unit(session):
    """Run the unit test suite."""

    session.install('coverage', 'pytest', 'pytest-cov', 'asyncmock', 'pytest-asyncio')
    session.install('-e', '.')

    session.run(
        'py.test',
        '--quiet',
        '--cov=google/cloud/gkeconnect/gateway_v1beta1/',
        '--cov-config=.coveragerc',
        '--cov-report=term',
        '--cov-report=html',
        os.path.join('tests', 'unit', ''.join(session.posargs))
    )


@nox.session(python='3.7')
def cover(session):
    """Run the final coverage report.
    This outputs the coverage report aggregating coverage from the unit
    test runs (not system test runs), and then erases coverage data.
    """
    session.install("coverage", "pytest-cov")
    session.run("coverage", "report", "--show-missing", "--fail-under=100")

    session.run("coverage", "erase")


@nox.session(python=['3.6', '3.7'])
def mypy(session):
    """Run the type checker."""
    session.install('mypy', 'types-pkg_resources')
    session.install('.')
    session.run(
        'mypy',
        '--explicit-package-bases',
        'google',
    )


@nox.session
def update_lower_bounds(session):
    """Update lower bounds in constraints.txt to match setup.py"""
    session.install('google-cloud-testutils')
    session.install('.')

    session.run(
        'lower-bound-checker',
        'update',
        '--package-name',
        PACKAGE_NAME,
        '--constraints-file',
        str(LOWER_BOUND_CONSTRAINTS_FILE),
    )


@nox.session
def check_lower_bounds(session):
    """Check lower bounds in setup.py are reflected in constraints file"""
    session.install('google-cloud-testutils')
    session.install('.')

    session.run(
        'lower-bound-checker',
        'check',
        '--package-name',
        PACKAGE_NAME,
        '--constraints-file',
        str(LOWER_BOUND_CONSTRAINTS_FILE),
    )

@nox.session(python='3.6')
def docs(session):
    """Build the docs for this library."""

    session.install("-e", ".")
    session.install("sphinx<3.0.0", "alabaster", "recommonmark")

    shutil.rmtree(os.path.join("docs", "_build"), ignore_errors=True)
    session.run(
        "sphinx-build",
        "-W",  # warnings as errors
        "-T",  # show full traceback on exception
        "-N",  # no colors
        "-b",
        "html",
        "-d",
        os.path.join("docs", "_build", "doctrees", ""),
        os.path.join("docs", ""),
        os.path.join("docs", "_build", "html", ""),
    )
